### PR TITLE
fix(storefront): BCTHEME-349 improve email validation for forms

### PR DESCRIPTION
**Pull request by bc-alexsaiannyi**
#### What?
This PR fixes an issue when User could fill in an email address with spaces on creating New Account or Account Settings pages. If the entered email  is not valid(spaces, line breaks and tabs) User can not submit a form. The video how it works now is attached to the main ticket.

#### Tickets / Documentation
- [BCTHEME-349](https://jira.bigcommerce.com/browse/BCTHEME-349)

#### Screenshots (if appropriate)

https://user-images.githubusercontent.com/67792608/113595290-7ed71e80-9641-11eb-9700-49d3b42bea74.mov

https://user-images.githubusercontent.com/67792608/113595438-ad54f980-9641-11eb-9be4-529182ab4983.mov

### CODE REVIEW

File: assets/js/theme/common/models/forms.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 const forms = {
     email(value) {
-        const re = /^.+@.+\..+/;
+        const re = /^\S+@\S+\.\S+/;
```
Comment (BC-tymurbiedukhin) [inline assets/js/theme/common/models/forms.js:3]: Why do we use capital S here?

Comment (bc-alexsaiannyi) [reply assets/js/theme/common/models/forms.js:3]: it is [used](https://www.w3schools.com/jsref/jsref_regexp_whitespace_non.asp) to exclude a non-whitespace characters in email

**Review by BC-tymurbiedukhin — APPROVED** [review]
LGTM

**Comment by yurytut1993** [comment]
@bc-alexsaiannyi @BC-tymurbiedukhin 
I believe we should use other regex to cover all invalid cases of email input
Screenshot 2021-04-06 at 18 18 52

**Comment by yurytut1993** [comment]
> @bc-alexsaiannyi @BC-tymurbiedukhin
> I believe we should use other regex to cover all invalid cases of email input
> Screenshot 2021-04-06 at 18 18 52

 on discussions decided to cover just invalid email spaces in this ticket

**Review by yurytut1993 — APPROVED** [review]

**Pull request merged**